circuits/src/zk_circuits/valid_commitments.rs

```diff
@@ -0,0 +1,950 @@
+
+        assert!(constraints_satisfied(witness, statement))
+    }
+
+    /// Tests the case in which an augmented balance must be added
+    #[test]
+    fn test_valid_commitments__valid_augmentation() {
+        let wallet = UNAUGMENTED_WALLET.clone();
+        let (witness, statement) = create_witness_and_statement(&wallet);
+
+        assert!(constraints_satisfied(witness, statement))
+    }
+
+    /// Tests the case in which the prover attempts to add a non-zero balance to the augmented wallet
+    #[test]
+    fn test_invalid_commitment__augmented_nonzero_balance() {
+        let wallet = UNAUGMENTED_WALLET.clone();
+        let (mut witness, statement) = create_witness_and_statement(&wallet);
+
+        // Prover attempt to augment the wallet with a non-zero balance
+        let augmented_balance_index = statement.balance_receive_index;
+        witness.augmented_public_shares.balances[augmented_balance_index].amount += Scalar::one();
+        witness.balance_receive.amount += 1u64;
+
+        assert!(!constraints_satisfied(witness, statement));
+    }
+
+    /// Tests the case in which the prover clobbers a non-zero balance to augment the wallet
+    #[test]
+    fn test_invalid_commitment__augmentation_clobbers_balance() {
+        let wallet = UNAUGMENTED_WALLET.clone();
+        let (mut witness, statement) = create_witness_and_statement(&wallet);
+
+        // Reset the original wallet such that the augmented balance was non-zero
+        let augmentation_index = statement.balance_receive_index;
+        witness.public_secret_shares.balances[augmentation_index] = BalanceSecretShare {
+            amount: Scalar::one(),
+            mint: Scalar::one(),
+        };
+
+        assert!(!constraints_satisfied(witness, statement))
+    }
+
+    /// Tests the case in which a prover attempts to modify an order in the augmented wallet
+    #[test]
+    fn test_invalid_commitment__augmentation_modifies_order() {
+        let wallet = UNAUGMENTED_WALLET.clone();
+        let (mut witness, statement) = create_witness_and_statement(&wallet);
+
+        // Modify an order in the augmented wallet
+        witness.augmented_public_shares.orders[1].amount += Scalar::one();
+
+        assert!(!constraints_satisfied(witness, statement));
+    }
+
+    /// Test the case in which a prover attempts to modify a fee in the augmented wallet
+    #[test]
+    fn test_invalid_commitment__augmentation_modifies_fee() {
+        let wallet = UNAUGMENTED_WALLET.clone();
+        let (mut witness, statement) = create_witness_and_statement(&wallet);
+
+        // Modify a fee in the wallet
+        witness.augmented_public_shares.fees[0].gas_token_amount += Scalar::one();
+        assert!(!constraints_satisfied(witness, statement));
+    }
+
+    /// Test the case in which a prover attempts to modify wallet keys and blinders in augmentation
+    #[test]
+    fn test_invalid_commitment__augmentation_modifies_keys_blinder() {
```

Comment:
nit: should this be split out into 2 tests?

Reply:
fixed